src/test/groovy/com/cloudbees/groovy/cps/CpsTransformerTest.groovy

```diff
@@ -556,4 +556,10 @@ class CpsTransformerTest extends AbstractGroovyCpsTest {
     void varArgs() {
         assert evalCPS('def fn(String... args) { args.size() }; fn("one string")') == 1;
     }
+
+    @Issue("JENKINS-28277")
+    @Test
+    void currying() {
+        assert evalCPS('def nCopies = { int n, String str -> str*n }; def twice=nCopies.curry(2); twice("foo")') == "foofoo";
```

Comment:
Would like to see a test `curry`ing a native method, as well as something in `DefaultGroovyMethods`.